tests/test_migrate_hardening.py

```
@@ -146,10 +147,11 @@ def test_database_migration_success(tmp_path: Path) -> None:
     assert migrated_db_path.exists()
     assert not db_path.exists()
 
-    with sqlite3.connect(migrated_db_path) as conn:
-        row = conn.execute(
-            "SELECT COUNT(*) FROM sqlite_master WHERE type='table' AND name='test'"
-        ).fetchone()
+    with contextlib.closing(sqlite3.connect(migrated_db_path)) as conn:
+        with conn as managed_conn:
```

Ruff (0.15.10)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)